bsmetadata/preprocessing_tools/html_parser/filters_and_cleaners.py

```diff
@@ -291,25 +300,48 @@ def __init__(
 
     def apply(self):
         html_str = self.html_str
-        # Traitement n°1: start the parsing at a special tags (mostly tested with <body>)
+        # Traitement n°1: isolate part of the DOM
+        additional_columns = {}
+        if self.tags_sub_tree_to_isolate:
+            try:
+                root = fromstring(html_str)
+            except ValueError:
+                # this is not a valid HTML (begin probably with <?xml version="1.0")
+                logger.warning(f"This example wasn't parsed: invalid HTML")
+                return "", [], {}
+            for tag in self.tags_sub_tree_to_isolate:
+                find = etree.XPath(f"//{tag}")
+                matches = find(root)
+                html_extracted = [
+                    etree.tostring(match, encoding="UTF-8", pretty_print=False).decode("UTF-8") for match in matches
+                ]
+                additional_columns[tag] = html_extracted
+
+        # Traitement n°2: start the parsing at a special tags (mostly tested with <body>)
         if self.start_parsing_at_tag is not None:
-            root = fromstring(html_str)
-            find = etree.XPath(f"//{self.start_parsing_at_tag}")
             try:
-                new_etree = find(root)[0]
-            except IndexError:
-                raise ValueError(
+                root = fromstring(html_str)
+            except ValueError:
+                # this is not a valid HTML (begin probably with <?xml version="1.0")
+                logger.warning(f"This example wasn't parsed: invalid HTML")
+                return "", [], {}
+            find = etree.XPath(f"//{self.start_parsing_at_tag}")
+            matches = find(root)
+            if len(matches) == 0:
+                logger.warning(
```

Comment:
These additions avoid two types of errors encountered:
- a file that the `lxml` library cannot parse as html
- the case where the html does not contain a "body" tag